api/v1alpha1/propagation_types.go

```diff
@@ -186,6 +216,32 @@ type Propagation struct {
 	Status PropagationStatus `json:"status,omitempty"`
 }
 
+func (p *Propagation) NextVersion() string {
+	versions := []string{}
+	if len(p.Status.DeploymentStatusesReports) == 0 {
+		return ""
+	}
+	statuses := p.Status.DeploymentStatusesReports[0].Statuses
+	for i := len(statuses) - 1; i >= 0; i-- {
+		version := statuses[i].Version
+		if !slices.Contains(versions, version) {
+			versions = append(versions, version)
+		}
+	}
+
+	now := time.Now()
+versions:
+	for _, v := range versions {
+		for _, r := range p.Status.DeploymentStatusesReports {
```

Comment:
I don't understand this logic. You iterate over statuses of all deployments? And as soon as this version doesn't have status (never deployed) in specific deployment, you jump immediately checking another version? That doesn't look right to me.